packages/api/src/controllers/helpers.ts

```diff
@@ -771,45 +771,49 @@ export async function addDefaultProjectId(
     }
   };
 
-  const enrichResponseWithUserProjectId = (document) => {
+  const enrichResponseWithUserProjectId = async (document) => {
     if ("id" in document && "userId" in document && "user" in document) {
-      if (
-        (!document.projectId || document.projectId === "") &&
-        document.user?.defaultProjectId
-      ) {
-        document.projectId = document.user.defaultProjectId;
+      if (!document.projectId || document.projectId === "") {
+        if (document.user?.defaultProjectId) {
+          document.projectId = document.user.defaultProjectId;
```

Comment:
We shouldn't be fetching the user from the DB again if the user is already embedded in the object.
```suggestion
        if (document.user) {
          document.projectId = document.user.defaultProjectId;
```